### PR TITLE
Separate shortcut building from hierarchy builder

### PR DESCRIPTION
**Pull request by dnesbitt61**

### CODE REVIEW

**Comment by kevinkreiser** [comment]
lots of change with no description!? my brain is mush.. is there anything important to know in here other than shortcuts got moved out of hierarchy building and we're building them on the fly without precomputing which nodes are 'contractable'? looks to me like not but just wanted to ask.

:rocket:

File: src/mjolnir/hierarchybuilder.cc [hunk 1]
```diff
@@ -584,8 +70,9 @@ std::pair<uint32_t, uint32_t> FormTilesInNewLevel(
     }
 
     // Check if we need to clear the tile cache
-    if(info.graphreader_.OverCommitted())
+    if (info.graphreader_.OverCommitted()) {
```
Comment (gknisely) [inline src/mjolnir/hierarchybuilder.cc]: if (meh) {
sigh {} for one line
}

**Comment by gknisely** [comment]
:shipit:

**Comment by dnesbitt61** [comment]
Separating the logic for building shortcuts simplifies the code within HierarchyBuilder. Also this now builds shortcuts on the fly, rather than gathering a list of nodes that can be contracted. This is still pretty fast, and simplifies the shortcut building slightly.

**Pull request merged**